### ISSUE TITLE
Graphical redesign of pagination

### ISSUE THREAD

**Issue by trang** [Tatoeba/tatoeba2#1134]
Current pagination:

![pagination](https://cloud.githubusercontent.com/assets/221850/15056068/25d2940e-1310-11e6-991d-1578b7728761.png)

Draft of new pagination:

![a834fc9e-f258-11e5-8578-232903b79918](https://cloud.githubusercontent.com/assets/221850/15056084/32ff2e3a-1310-11e6-8f33-1374f83cff02.png)

This needs to be done before #24.

**Comment by RyckRichards** [comment]
It would be great to have an "extra field" which would allow users to
type the page they want to go to and both a "Go" icon and an
"information" that he can also press Enter. It would save us a lot of
time when we search some sorts of sentences, for example.

On 5/5/16, Trang <notifications@github.com> wrote:
> Current pagination:
>
> ![pagination](https://cloud.githubusercontent.com/assets/221850/15056068/25d2940e-1310-11e6-991d-1578b7728761.png)
>
> Draft of new pagination:
>
> ![a834fc9e-f258-11e5-8578-232903b79918](https://cloud.githubusercontent.com/assets/221850/15056084/32ff2e3a-1310-11e6-8f33-1374f83cff02.png)
>
>
> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Tatoeba/tatoeba2/issues/1134


-- 
Ricardo Vernaut Jr

**Comment by trang** [comment]
@RyckRichards, this is what #24 is about.

**Comment by RyckRichards** [comment]
Oh sorry...

-- 
Ricardo Vernaut Jr

**Comment by billoliver** [comment]
### The Redesign
![pagination-tatoeba](https://cloud.githubusercontent.com/assets/17387379/15274493/60f43cba-1a89-11e6-8987-ce9bf7ce7b5d.png)

- I increased both the size of clickable elements and the space between each page link so that users can't accidentally click on the wrong one.

- Only the current page link has a background color so that users can easily keep track of what page they are.

- The first and last page icon was removed because some users might think they will take them to the next page. If a user wants to go to those pages he or she just need to click the links.

**Colors:**
```
Chevron icons: #000
Current page background: #4caf50
Current page font size: 14px
Current page font weight: bold
Current page element size: 24px | 24px
Default color of the links: #757575
Hover: #eee
```

**Comment by trang** [comment]
Thank you @billoliver! I've posted a comment on the Wall to ask for feedback.
https://tatoeba.org/eng/wall/show_message/26409#message_26409

Just one detail: the current pagination has 7 elements displayed (if we don't count the "first", "last", "previous" and next "elements"), any reason why you reduced it to 6? 

The reason why there are currently 7 elements is because we display pages -3 and +3 from the selected page. With 6 elements we don't have this symmetry anymore. 
If you think 7 elements is too much, we can reduce it to 5, so that we can still keep the symmetry, with the pages -2 and +2 from the selected page.

**Comment by ckjpn** [comment]
One possibility is to only display ever other number.  Maybe it's not a great idea.  I'm just brainstorming.

<  4  6 [8] 10 12 >

[8] being the current page.

With the Previous and Next links, at least, you don't really need the numbers for the 2 pages immediately before and after the current page.

Another idea that would allow people to jump to pages further away from the current page would be to calculate how many pages there are, then divide that number by the number of desired page links, and display equally-space page numbers.

What I mean is something like this.

http://www.manythings.org/bilingual/tur/164.html
![screen shot 2016-05-16 at 09 24 25](https://cloud.githubusercontent.com/assets/7934273/15277773/0dd491d2-1b48-11e6-85a1-9613b01ba080.png)

**Comment by ckjpn** [comment]
This is also just a "brainstorming" idea. I don't know if you would consider it appropriate for tatoeba.org.

You might want to consider adding a "random page" link, somewhat similar to this.

![screen shot 2016-05-16 at 09 59 19](https://cloud.githubusercontent.com/assets/7934273/15278001/135ccb2e-1b4d-11e6-83a1-6d40e858375a.png)

http://study.aitech.ac.jp/audioplayer/por/5113.html

**Comment by sabretou** [comment]
I second the random page idea. That sounds like it could get a lot of use.

Are the ellipses (...) clickable? If so, what happens when you click on them? I'd expect a little pop-up to appear, asking what page the user wants to go to. This way, users can quickly jump to, say, page 70 if they wanted to.

**Comment by ckjpn** [comment]
I do like the bigger buttons like this idea.

https://cloud.githubusercontent.com/assets/221850/15056084/32ff2e3a-1310-11e6-8f33-1374f83cff02.png

It's likely to make it much easier to use the website on smaller touchscreen devices.

**Comment by billoliver** [comment]
> Are the ellipses (...) clickable?

Yes, it is.

> I second the random page idea. That sounds like it could get a lot of use.

You guys might want to take a look at this issue #1063

**Comment by Guybrush88** [comment]
@billoliver what about leaving the quick jump to the first and the last page and adding a popup message saying like "jump directly to the first/last page" (or whatever) when someone puts the cursor on the relative icons?

**Comment by billoliver** [comment]
@Guybrush88 Users can already do that by clicking the following links:

![guybrush](https://cloud.githubusercontent.com/assets/17387379/15295841/02a43d96-1b69-11e6-9f64-5e2cc5e0f11b.png)

**Comment by trang** [comment]
@billoliver, I didn't think the ellipsis were clickable. What do they do?

@sabretou and @ckjpn in this issue we're only dealing with a graphical redesign of the pagination, we're not adding new features, just making it look different than it looks now. 
Having a "random page" button is a possible idea, but we should discuss it in another issue. If you were ever in a situation where you felt it would be nice to have a random page button on Tatoeba, feel free to create an issue and describe the situation.

**Comment by billoliver** [comment]
@trang It will show this feature #24

**Comment by trang** [comment]
Okay I see. So it wouldn't be clickable at first, but it will/may be when we implement #24.

Another question: since now the background is grey, what would be the "hover" color?

**Comment by billoliver** [comment]
> Another question: since now the background is grey, what would be the "hover" color?

A darker gray: #e0e0e0

**Comment by ckjpn** [comment]
If you want to play around with the CSS to just manipulate how the current pages look, you can experiment with these.

Download: http://www.manythings.org/tatoeba/paging-test-pages.zip

These are the pages and screenshots

paging-css-001-original.html
This is the original CSS for the "paging", but I think perhaps one other piece of CSS from somewhere else affects how this looks on the main site.

![screen shot 2016-05-19 at 16 51 41](https://cloud.githubusercontent.com/assets/7934273/15386464/fcd4e9ca-1de2-11e6-91a6-b12f635f1647.png)

paging-css-002-border.html

![screen shot 2016-05-19 at 16 51 49](https://cloud.githubusercontent.com/assets/7934273/15386487/1eedcf40-1de3-11e6-882e-ab4df8fafcb2.png)

paging-css-003-more-padding-less-margin.html

![screen shot 2016-05-19 at 16 51 56](https://cloud.githubusercontent.com/assets/7934273/15386493/24f6ee58-1de3-11e6-8a2f-dc3e7699e0f9.png)

paging-css-004-backgound-color-change.html

![screen shot 2016-05-19 at 16 52 03](https://cloud.githubusercontent.com/assets/7934273/15386543/59d0b618-1de3-11e6-9a41-5bf2cee69dd6.png)

paging-css-005-link-color-change.html

![screen shot 2016-05-19 at 16 52 14](https://cloud.githubusercontent.com/assets/7934273/15386549/621af374-1de3-11e6-9862-c738aed7eee1.png)

paging-css-006-link-more-padding.html

![screen shot 2016-05-19 at 16 52 26](https://cloud.githubusercontent.com/assets/7934273/15386553/682b6e9c-1de3-11e6-8ccd-df8c56ec8d48.png)

**Comment by ckjpn** [comment]
At the same time you are thinking of updating this "look", perhaps you should make sure the the "previous" "next" and "random" link on the "sentence" pages have the same "look", so everything is standard.

https://tatoeba.org/eng/sentences/show/1187912

![screen shot 2016-05-19 at 18 57 01](https://cloud.githubusercontent.com/assets/7934273/15389852/da3b010e-1df3-11e6-8f14-2bc3f97070d5.png)

**Comment by trang** [comment]
@ckjpn those are not exactly pagination items. They definitely need to be redesigned but that's out of scope for this issue. 

The design for the pagination is somewhat stable now, we won't need to experiment much more. To summarize, the new pagination will look more or less like this:

![guybrush](https://cloud.githubusercontent.com/assets/17387379/15295841/02a43d96-1b69-11e6-9f64-5e2cc5e0f11b.png)

**Icons**
* https://design.google.com/icons/#ic_more_horiz
* https://design.google.com/icons/#ic_keyboard_arrow_left
* https://design.google.com/icons/#ic_keyboard_arrow_right

**Colors**
* Background & text color of selected page: `#66AA00` & `#FFFFFF`
* Background & text color of unselected pages: `#EEEEEE` & `#7D7D7D`
* Background & text color when hover on unselected pages: `#E0E0E0` & `#7D7D7D`
* Color of enabled & disabled arrow icons: `#000000` & `#BDBDBD`
* Color of `...` icon: `#BDBDBD`

**Disabled / not displayed**
* If the selected page is less or equal to 3
  - the `<` icon is disabled 
  - the `[1]` page item is not displayed
  - the left-side `...` icon is not displayed
* If the selected page is one of the 3 last pages
  - the `>` icon is disabled 
  - the `[{last-page}]` page item is not displayed
  - the right-side `...` icon is not displayed

**Margins, paddings, font size**
That's left to the developer to evaluate, we don't have to be pixel perfect there. It should just look approximately like the mockup.

@billoliver, maybe you can confirm the colors.

**Comment by billoliver** [comment]
I've updated the colors.
![new-pagination](https://cloud.githubusercontent.com/assets/17387379/15402272/c8b66f9c-1dc9-11e6-9f67-bd88294c28f7.png)

**Colors**

- Background & text color of selected page: `#66AA00` & `#FFFFFF`
- Background & text color of unselected pages: `#EEEEEE` & **`#757575`**
- Background & text color when hover on unselected pages: #E0E0E0 & **`#757575`**
- Color of enabled & disabled arrow icons: #000000 & #BDBDBD
- Color of ... icon: #BDBDBD
